### PR TITLE
API Reference Template | add inherit info & use macros for the method

### PR DESCRIPTION
**Pull request by skyclouds2001**
### Description

this add the sentence suggested in https://github.com/mdn/content/pull/30946#issuecomment-1857214538 and https://github.com/mdn/content/pull/30946#issuecomment-1857513693, which helps express inherits information of events

this is useful for many DOM interfaces (`HTMLElement` interface and so on), also some other interface like `ScreenDetailed`

also, replace the `addEventListener()` method with macros

### CODE REVIEW

**Comment by zfox23** [comment]
Thank you! This looks fine to me, but I'd prefer if @teoli2003 and/or @wbamberg weighed in, since they were the folks who authored the two comments you linked.

**Review by wbamberg — APPROVED** [review]
👍 thank you for the PR, @skyclouds2001 , and for the ping, @zfox23 ! I think this is great. Also tagging @hamishwillee , who's involved in this conversation.

File: files/en-us/mdn/writing_guidelines/page_structures/page_types/api_reference_page_template/index.md [hunk 1]
```diff
@@ -120,7 +120,9 @@ Include one term and definition for each method.
 
 ## Events
 
-Listen to these events using [`addEventListener()`](/en-US/docs/Web/API/EventTarget/addEventListener) or by assigning an event listener to the `oneventname` property of this interface.
+Listen to these events using {{DOMxRef("EventTarget.addEventListener", "addEventListener()")}} or by assigning an event listener to the `oneventname` property of this interface.
+
+_Also inherits events from its parent interface, `\{{DOMxRef("NameOfParentInterface")}}`._ (Note: If the interface doesn't inherit from another interface, remove this whole line.)
```
Comment (hamishwillee) [inline files/en-us/mdn/writing_guidelines/page_structures/page_types/api_reference_page_template/index.md:125]: Reorder for consistency (IMO)

```suggestion
_Also inherits events from its parent interface, `\{{DOMxRef("NameOfParentInterface")}}`._ (Note: If the interface doesn't inherit from another interface, remove this whole line.)

Listen to these events using {{DOMxRef("EventTarget.addEventListener", "addEventListener()")}} or by assigning an event listener to the `oneventname` property of this interface.
```

**Review by hamishwillee — APPROVED** [review]
Thanks. I've put the "Also inherits ..." first, for consistency with the other headings. Otherwise great.

@skyclouds2001 Thanks for driving this!

**Review by teoli2003 — APPROVED** [review]
👍 Also approving.

**Comment by hamishwillee** [comment]
That's enough of us. Merging. Thank you @skyclouds2001 !

**Pull request merged**